lib/testScenario/postmanCollectionRunnerClient.ts

```diff
@@ -0,0 +1,437 @@
+      method: step.operation._method as string,
+      url: "",
+      body: { mode: "raw" } as RequestBodyDefinition,
+    });
+    const queryParams: QueryParamDefinition[] = [];
+    const urlVariables: VariableDefinition[] = [];
+    for (const p of step.operation.parameters ?? []) {
+      const param = this.jsonLoader.resolveRefObj(p);
+      const paramValue = stepEnv.env.get(param.name) || step.exampleTemplate.parameters[param.name];
+      if (!this.collectionEnv.has(param.name)) {
+        this.collectionEnv.set(
+          param.name,
+          paramValue,
+          typeof step.exampleTemplate.parameters[param.name]
+        );
+      }
+
+      switch (param.in) {
+        case "path":
+          urlVariables.push({ key: param.name, value: `{{${param.name}}}` });
+          break;
+        case "query":
+          queryParams.push({ key: param.name, value: paramValue });
+          break;
+        case "header":
+          const header = new Header({ key: param.name, value: paramValue });
+          item.request.headers.add(header);
+          break;
+        case "body":
+          item.request.body = new RequestBody({
+            mode: "raw",
+            raw: JSON.stringify(_request.body, null, 2),
+          });
+          break;
+        default:
+          throw new Error(`Parameter "in" not supported: ${param.in}`);
+      }
+      this.collection.items.add(item);
+    }
+    const authorizationHeader = new Header({
+      key: "Authorization",
+      value: `Bearer {{bearerToken}}`,
+    });
+    const contentType = new Header({ key: "Content-Type", value: "application/json" });
+    item.request.addHeader(contentType);
+    item.request.addHeader(authorizationHeader);
+
+    // store example variable. generate auto validation
+    this.addTestScript(item);
+    item.request.url = new Url({
+      path: pathEnv.resolveString(step.operation._path._pathTemplate, "{", "}"),
+      host: "https://management.azure.com",
+      variable: urlVariables,
+    } as UrlDefinition);
+    item.request.addQueryParams(queryParams);
+
+    if (step.operation["x-ms-long-running-operation"]) {
+      this.addAsLongRunningOperationItem(item);
+    } else {
+      this.collection.items.add(item);
+    }
+    if (step.operation._method === "put" || step.operation._method === "delete") {
+      this.collection.items.add(
+        this.getOperationItem(
+          `${item.name}_${step.operation._method}_generated`,
+          item.request.url.toString()
+        )
+      );
+    }
+  }
+
+  private addAsLongRunningOperationItem(item: Item) {
+    this.collectionEnv.set(`${item.name}_polling_url`, "<polling_url>", "string");
+    const longRunningEvent = new Event({
+      listen: "test",
+      script: {
+        type: "text/javascript",
+        exec: `pm.environment.set("${item.name}_polling_url", pm.response.headers.get('Location')||pm.response.headers.get('Azure-AsyncOperation')||"https://postman-echo.com/delay/10")`,
```

Comment:
Header may be lower cased. And you need to support Operation-Location

Reply:
What's Operation-Location?